devices/Sps30/Entities/Measurement.cs

```diff
@@ -0,0 +1,124 @@
+        /// requested type is deducted and parsed accordingly.
+        /// </summary>
+        /// <param name="data">The response data on the requested measurement</param>
+        public Measurement(byte[] data)
+        {
+            if (data.Length >= 40)
+            {
+                // When we have 40 bytes of data, we assume Float was requested and will be parsed as such
+                Format = MeasurementOutputFormat.Float;
+                MassConcentrationPM10 = BigEndianBitConverter.ToSingle(data, 0);
+                MassConcentrationPM25 = BigEndianBitConverter.ToSingle(data, 4);
+                MassConcentrationPM40 = BigEndianBitConverter.ToSingle(data, 8);
+                MassConcentrationPM100 = BigEndianBitConverter.ToSingle(data, 12);
+                NumberConcentrationPM05 = BigEndianBitConverter.ToSingle(data, 16);
+                NumberConcentrationPM10 = BigEndianBitConverter.ToSingle(data, 20);
+                NumberConcentrationPM25 = BigEndianBitConverter.ToSingle(data, 24);
+                NumberConcentrationPM40 = BigEndianBitConverter.ToSingle(data, 28);
+                NumberConcentrationPM40 = BigEndianBitConverter.ToSingle(data, 32);
+                TypicalParticleSize = BigEndianBitConverter.ToSingle(data, 36);
+            }
+            else if (data.Length >= 20)
+            {
+                // When we have 20 bytes of data, we assume UInt16 was requested and will be parsed as such
+                Format = MeasurementOutputFormat.UInt16;
+                MassConcentrationPM10 = BigEndianBitConverter.ToUInt16(data, 0);
+                MassConcentrationPM25 = BigEndianBitConverter.ToUInt16(data, 2);
+                MassConcentrationPM40 = BigEndianBitConverter.ToUInt16(data, 4);
+                MassConcentrationPM100 = BigEndianBitConverter.ToUInt16(data, 6);
+                NumberConcentrationPM05 = BigEndianBitConverter.ToUInt16(data, 8);
+                NumberConcentrationPM10 = BigEndianBitConverter.ToUInt16(data, 10);
+                NumberConcentrationPM25 = BigEndianBitConverter.ToUInt16(data, 12);
+                NumberConcentrationPM40 = BigEndianBitConverter.ToUInt16(data, 14);
+                NumberConcentrationPM40 = BigEndianBitConverter.ToUInt16(data, 16);
+                TypicalParticleSize = BigEndianBitConverter.ToUInt16(data, 18);
+            }
+            else
+            {
+                throw new ApplicationException($"Not enough bytes received to parse a measurement");
+            }
+        }
+
+        /// <summary>
+        /// The format assumed when parsing the data for this measurement instance.
+        /// </summary>
+        public MeasurementOutputFormat Format { get; protected set; }
+
+        /// <summary>
+        /// Mass Concentration PM1.0 [µg/m³]
+        /// </summary>
+        public double MassConcentrationPM10 { get; protected set; }
```

Comment:
For the abbreviations, it should be Pm10
Note: to be consistent with the rest of the bindings and .NET IoT, we need to add the UnitsNet for MassConcentration. The nuget is not available yet. I can make the PR and it will be in few days. OK to wait for it to be present?
(there is none for number per volume)

Reply:
And sorry it's still not there :-(

Reply:
It's there! :)